<a href="https://colab.research.google.com/github/Mr-Himanshu123/Fraud-Detection/blob/main/Fraud_Detection.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## **Dataset Description :-**

step - maps a unit of time in the real world. In this case 1 step is 1 hour of time. Total steps 744 (30 days simulation).

type - CASH-IN, CASH-OUT, DEBIT, PAYMENT and TRANSFER.

amount - amount of the transaction in local currency.

nameOrig - customer who started the transaction

oldbalanceOrg - initial balance before the transaction

newbalanceOrig - new balance after the transaction

nameDest - customer who is the recipient of the transaction

oldbalanceDest - initial balance recipient before the transaction. Note that there is not information for customers that start with M (Merchants).

newbalanceDest - new balance recipient after the transaction. Note that there is not information for customers that start with M (Merchants).

isFraud - This is the transactions made by the fraudulent agents inside the simulation. In this specific dataset the fraudulent behavior of the agents aims to profit by taking control or customers accounts and try to empty the funds by transferring to another account and then cashing out of the system.

isFlaggedFraud - The business model aims to control massive transfers from one account to another and flags illegal attempts. An illegal attempt in this dataset is an attempt to transfer more than 200.000 in a single transaction.

In [23]:
df.head()

step      type    amount     nameOrig  oldbalanceOrg  newbalanceOrig  \
0     1   PAYMENT   9839.64  C1231006815       170136.0       160296.36   
1     1   PAYMENT   1864.28  C1666544295        21249.0        19384.72   
2     1  TRANSFER    181.00  C1305486145          181.0            0.00   
3     1  CASH_OUT    181.00   C840083671          181.0            0.00   
4     1   PAYMENT  11668.14  C2048537720        41554.0        29885.86   

      nameDest  oldbalanceDest  newbalanceDest  isFraud  isFlaggedFraud  
0  M1979787155             0.0             0.0      0.0             0.0  
1  M2044282225             0.0             0.0      0.0             0.0  
2   C553264065             0.0             0.0      1.0             0.0  
3    C38997010         21182.0             0.0      1.0             0.0  
4  M1230701703             0.0             0.0      0.0             0.0

In [3]:
#Importing libraries
import pandas as pd
import numpy as np

In [1]:
#Mounting a drive to load dataset
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [4]:
# data loading
df = pd.read_csv('/content/drive/MyDrive/Insance project/Fraud.csv.crdownload')

In [5]:
#First look
df.head()

step      type    amount     nameOrig  oldbalanceOrg  newbalanceOrig  \
0     1   PAYMENT   9839.64  C1231006815       170136.0       160296.36   
1     1   PAYMENT   1864.28  C1666544295        21249.0        19384.72   
2     1  TRANSFER    181.00  C1305486145          181.0            0.00   
3     1  CASH_OUT    181.00   C840083671          181.0            0.00   
4     1   PAYMENT  11668.14  C2048537720        41554.0        29885.86   

      nameDest  oldbalanceDest  newbalanceDest  isFraud  isFlaggedFraud  
0  M1979787155             0.0             0.0      0.0             0.0  
1  M2044282225             0.0             0.0      0.0             0.0  
2   C553264065             0.0             0.0      1.0             0.0  
3    C38997010         21182.0             0.0      1.0             0.0  
4  M1230701703             0.0             0.0      0.0             0.0

In [12]:
#checking the shape of the dataset
df.shape

(1162838, 11)

In [6]:
#Info of the data
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1162838 entries, 0 to 1162837
Data columns (total 11 columns):
 #   Column          Non-Null Count    Dtype  
---  ------          --------------    -----  
 0   step            1162838 non-null  int64  
 1   type            1162838 non-null  object 
 2   amount          1162838 non-null  float64
 3   nameOrig        1162838 non-null  object 
 4   oldbalanceOrg   1162838 non-null  float64
 5   newbalanceOrig  1162838 non-null  float64
 6   nameDest        1162838 non-null  object 
 7   oldbalanceDest  1162837 non-null  float64
 8   newbalanceDest  1162837 non-null  float64
 9   isFraud         1162837 non-null  float64
 10  isFlaggedFraud  1162837 non-null  float64
dtypes: float64(7), int64(1), object(3)
memory usage: 97.6+ MB


In [15]:
#checking duplicate value
df.duplicated().sum()

0

In [16]:
#Checking null value
df.isnull().sum()

step              0
type              0
amount            0
nameOrig          0
oldbalanceOrg     0
newbalanceOrig    0
nameDest          0
oldbalanceDest    1
newbalanceDest    1
isFraud           1
isFlaggedFraud    1
dtype: int64

In [17]:
#Checking uniques of each variable
df.nunique()

step                  132
type                    5
amount            1115524
nameOrig          1162523
oldbalanceOrg      451619
newbalanceOrig     515911
nameDest           500417
oldbalanceDest     667490
newbalanceDest     507294
isFraud                 2
isFlaggedFraud          1
dtype: int64

In [22]:
#Checking values counts of important variable
print('type _________________')
print(df['type'].value_counts())
print('\n')

print('isFraud _______________')
print(df['isFraud'].value_counts())
print('\n')

print('isFlaggedFraud___________')
print(df['isFlaggedFraud'].value_counts())

type _________________
CASH_OUT    411302
PAYMENT     394584
CASH_IN     252523
TRANSFER     96377
DEBIT         8052
Name: type, dtype: int64


isFraud _______________
0.0    1161329
1.0       1508
Name: isFraud, dtype: int64


isFlaggedFraud___________
0.0    1162837
Name: isFlaggedFraud, dtype: int64


### **Insights we found from the above analysis : -**
1. Based on the provided data, it can be observed that there are 1,162,838 rows and 11 columns.
2. No duplicate values are present in the dataset.
3. The columns 'oldbalanceDest,' 'newbalanceDest,' 'isFraud,' and 'isFlaggedFraud' each contain only one missing value. As a result, we can safely drop these columns in the further analysis.
4.The 'isFlaggedFraud' column contains only a single type of data, indicating that it has no variability. Therefore, it is unnecessary for our analysis, and we can safely exclude this column.